### PR TITLE
docs(i-made-this): added new article on idempotency with CDK and Powertools 

### LINKED ISSUE #2568: [I Made This]: idempotency with cdk and powertools
### Link to your material

https://www.ranthebuilder.cloud/post/serverless-api-idempotency-with-aws-lambda-powertools-and-cdk

### Description

In this post, you will learn about serverless API idempotency, why it's essential, and how to implement it in a sample serverless "orders" service with AWS Lambda functions, AWS CDK, and AWS Lambda Powertools for Python. We will also write an end-to-end test to validate that API idempotency works as expected.

### Preferred contact

I'll add in the pr

### (Optional) Social Network

_No response_

### (Optional) Additional notes

_No response_

### Acknowledgment

- [X] I understand this content may be removed from Powertools for AWS Lambda (Python) documentation if it doesn't conform with the [Code of Conduct](https://aws.github.io/code-of-conduct)

### PR DESCRIPTION
**Pull request by ran-isenberg**
**Issue number:#2568**

## Summary
add idempotency blog post to collection
### Changes

> Please provide a summary of what's being changed

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [ ] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [ ] I have performed a self-review of this change
* [ ] Changes have been tested
* [ ] Changes are documented
* [ ] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

**Comment by ran-isenberg** [comment]
see https://github.com/aws-powertools/powertools-lambda-python/issues/2568

**Comment by rubenfonseca** [comment]
Looking at this now

**Comment by rubenfonseca** [comment]
Hi @ran-isenberg great article! Can I just ask you to change all links to Powertools docs to use the prefix `https://docs.powertools.aws.dev/lambda-python/latest/utilities/idempotency/`?

It was not your fault, we've just been messing with the URLs, but that prefix should be final.

Let me know when you finish it and I'll be happy to merge and publish!

File: docs/we_made_this.md [hunk 1]
```diff
@@ -35,6 +35,8 @@ A collection of articles explaining in detail how Lambda Powertools helps with a
 
 * [Part 6 - Configuration & Feature Flags](https://www.ranthebuilder.cloud/post/aws-lambda-cookbook-part-6-feature-flags-configuration-best-practices){:target="_blank"}
 
+* [Serverless API Idempotency with AWS Lambda Powertools and CDK](https://www.ranthebuilder.cloud/post/serverless-api-idempotency-with-aws-lambda-powertools-and-cdk){:target="_blank"}
```
Comment (rubenfonseca) [inline docs/we_made_this.md:38]: Hi @ran-isenberg great article! Can I just ask you to change all links to Powertools docs to use the prefix https://docs.powertools.aws.dev/lambda-python/latest/utilities/idempotency/?

It was not your fault, we've just been messing with the URLs, but that prefix should be final.

Let me know when you finish it and I'll be happy to merge and publish!

**Review by rubenfonseca — CHANGES_REQUESTED** [review]

File: docs/we_made_this.md (hunk 1 above)
Comment (ran-isenberg) [reply docs/we_made_this.md:38]: @rubenfonseca , I already have last week. Changed 13 articles. I have [ahref](https://ahrefs.com/) automation which found me all the broken links.

**Review by rubenfonseca — APPROVED** [review]

**Pull request merged**